### PR TITLE
feat(discover): Add model for saved queries

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: src/sentry/models/discoversavedquery.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+
+class DiscoverSavedQuery(Model):
+    """
+    A saved Discover query
+    """
+    __core__ = False
+
+    organization_id = BoundedPositiveIntegerField(db_index=True)
```
Comment (markstory) [inline src/sentry/models/discoversavedquery.py]: Should this model have associations to the Organization?

File: src/sentry/models/discoversavedquery.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    name = models.CharField(max_length=64)
+    project_ids = ArrayField(BoundedPositiveIntegerField())
```
Comment (markstory) [inline src/sentry/models/discoversavedquery.py]: Can anyone in a project access saved discover queries?

**Review by mattrobenolt — COMMENTED** [review]
This needs to get added into deletions code to be deleted when an Organization is deleted, otherwise we'll get rows that never go away.

And, is it valuable to have a "saved query" spelled out in the schema? Or should we just have a `data` blob that contains a JSON serialized query.

I ask because this seems like we're already using a lot of nested datastructures, and this seems pretty fragile with something that might need changing often with little value in having this in a format like this.

Maybe just blast out a `JSONField` for all of the bits?

**Comment by mattrobenolt** [comment]
As we discussed, we should also move `project_ids` column into a real M2M relationship.

And add `date_created/date_modified` columns.

**Comment by lynnagara** [comment]
@macqueen Can you look over this, especially the deletions code?

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**